### ISSUE TITLE
Unable to run project on Eclipse

### ISSUE THREAD

**Issue by kimbaudi** [spring-projects/spring-petclinic#222]
Hi, I am unable to run this project on Eclipse using Tomcat server. I found this closed [issue #128](https://github.com/spring-projects/spring-petclinic/issues/128) that describes how to run this project on Eclipse and tried to follow the steps. 

The first issue I came across when I imported this repo as an Existing Maven Project was that the Eclipse project folder structure did not show Deployment Descriptor and there was no Run As -> Run on Server option.

![screen shot 2016-12-25 at 7 39 19 am](https://cloud.githubusercontent.com/assets/1864092/21472076/91a9bf48-ca7d-11e6-9548-4f2d4d0a94fe.png)

So I did Run As -> Maven Install to install dependencies. And went to Project Properties -> Project Facets and selected Dynamic Web Modules to convert Maven project to Dynamic Web Project. Now, I could see the Deployment Descriptor and Run on Server Option.

![screen shot 2016-12-25 at 8 55 32 am](https://cloud.githubusercontent.com/assets/1864092/21472134/f3d25444-ca7f-11e6-922f-f4ba952a5126.png)

However, when I Run on Server -> Tomcat 8.5, I get a 404 error

![screen shot 2016-12-25 at 8 57 07 am](https://cloud.githubusercontent.com/assets/1864092/21472146/8f6c1ebc-ca80-11e6-9a00-ed25302b372e.png)

Any idea what I'm doing wrong? I've been at this for hours and don't have a clue. Can you provide me with instructions on getting this project on Eclipse and run on Tomcat server? I also have IntelliJ Ultimate, but I had problems trying to create Run Configuration on that IDE so I tried to do it in Eclipse. I'm not very experienced with either IDEs.

I am on MacOS Sierra using Eclipse Java EE IDE for Web Developers (Version: Neon.2 Release (4.6.2)) and Tomcat 8.5

**Comment by swapy** [comment]
This project has embedded tomcat . It is meant to be run as java application.

**Comment by kimbaudi** [comment]
Thanks for the quick reply. It helped me run this project on Eclipse.

I imported this project on Eclipse as a Maven project. Then I right-clicked the project folder and go to Run -> Run as Java Application. I got a prompt to Select Java Application:

![screen shot 2016-12-25 at 5 59 35 pm](https://cloud.githubusercontent.com/assets/1864092/21473886/555bcee2-cacc-11e6-9570-56765d60c31b.png)

I selected `Tomcat$1` because it is first on the list and I don't know exactly what I'm doing. Then I get this error:

```terminal
Error: A JNI error has occurred, please check your installation and try again
Usage: RealmBase [-a <algorithm>] [-e <encoding>] [-i <iterations>] [-s <salt-length>] [-k <key-length>] [-h <handler-class-name>] <credentials>
```

So I tried again, but this time I selected `PetClinicApplication` and it worked:

![screen shot 2016-12-25 at 6 15 35 pm](https://cloud.githubusercontent.com/assets/1864092/21474012/50695f92-cace-11e6-9744-c885ce7f44f0.png)

Now I can make changes to the project src and when I go to `localhost:8080` and the project is running on Tomcat with the changes I made. For example, I changed the "Welcome" text to "Howdy":

![screen shot 2016-12-25 at 6 59 34 pm](https://cloud.githubusercontent.com/assets/1864092/21474460/6361a45a-cad4-11e6-8639-a285a77bba1b.png)
